controller/hydrator/hydrator.go

```diff
@@ -241,6 +248,17 @@ func (h *Hydrator) getRelevantAppsForHydration(logCtx *log.Entry, hydrationKey t
 			continue
 		}
 
+		path := app.Spec.SourceHydrator.SyncSource.Path
+		// ensure that the path is always set to a path that doesn't resolve to the root of the repo
+		if IsRootPath(path) {
+			logCtx.Warnf(
+				"App %q has SyncSource.Path %q which resolves to the repository root. "+
+					"SyncSource.Path must point to a subdirectory within the repository, not the root.",
+				app.QualifiedName(), path,
+			)
```

Comment:
I don't think there's a need to log, the error will end up being logged higher up the chain.